message_ix_models/model/material/data_generic.py

```diff
@@ -40,7 +40,7 @@ def read_data_generic(scenario):
     return data_generic, data_generic_ts
 
 
-def gen_data_generic(scenario, dry_run=False):
+def gen_data_generic(scenario, dry_run=False) -> dict[str, pd.DataFrame]:
```

Comment:
```suggestion
def gen_data_generic(scenario: Scenario, dry_run: bool = False) -> dict[str, pd.DataFrame]:
```
The more type hints we use, the better the feature becomes. This might require `from message_ix import make_df, Scenario`.